### PR TITLE
impl(pubsub): add tracing pull ack handler

### PR DESCRIPTION
**Pull request by alevenberg**
Part of https://github.com/googleapis/google-cloud-cpp/issues/13265

2/5 in a chain of PRs to add tracing for the settle span for the unary pull

1. https://github.com/googleapis/google-cloud-cpp/pull/13350
2. https://github.com/googleapis/google-cloud-cpp/pull/13353
3. https://github.com/googleapis/google-cloud-cpp/pull/13354
4. https://github.com/googleapis/google-cloud-cpp/pull/13358
5. https://github.com/googleapis/google-cloud-cpp/pull/13360

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13353)

### CODE REVIEW

File: google/cloud/internal/opentelemetry.h [hunk 1]
```diff
@@ -94,6 +95,19 @@ opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span> MakeSpan(
     opentelemetry::trace::StartSpanOptions const& options =
         DefaultStartSpanOptions());
 
+/**
+ * Start a span with a @p name and @p attributes using templates.
```
Comment (coryan) [inline google/cloud/internal/opentelemetry.h:99]: I am not use "using templates" adds a lot.

File: google/cloud/internal/opentelemetry.h [hunk 2]
```diff
@@ -94,6 +95,19 @@ opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span> MakeSpan(
...
+ */
+template <class T>
```
Comment (coryan) [inline google/cloud/internal/opentelemetry.h:101]: We would typically say `typename T`.

File: google/cloud/pubsub/internal/tracing_pull_ack_handler.h [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_PUBSUB_INTERNAL_TRACING_PULL_ACK_HANDLER_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_PUBSUB_INTERNAL_TRACING_PULL_ACK_HANDLER_H
+
+#include "google/cloud/pubsub/pull_ack_handler.h"
+#include "google/cloud/pubsub/subscription.h"
+#include "google/cloud/pubsub/version.h"
+#include <memory>
+#include <string>
```
Comment (coryan) [inline google/cloud/pubsub/internal/tracing_pull_ack_handler.h:22]: nit: unused

File: google/cloud/pubsub/internal/tracing_pull_ack_handler.h [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_PUBSUB_INTERNAL_TRACING_PULL_ACK_HANDLER_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_PUBSUB_INTERNAL_TRACING_PULL_ACK_HANDLER_H
+
+#include "google/cloud/pubsub/pull_ack_handler.h"
+#include "google/cloud/pubsub/subscription.h"
```
Comment (coryan) [inline google/cloud/pubsub/internal/tracing_pull_ack_handler.h:19]: nit: unused

**Review by coryan — APPROVED** [review]

File: google/cloud/internal/opentelemetry.h (hunk 1 above)
Comment (alevenberg) [reply google/cloud/internal/opentelemetry.h]: ack, removed.

File: google/cloud/internal/opentelemetry.h (hunk 2 above)
Comment (alevenberg) [reply google/cloud/internal/opentelemetry.h]: done, and done for other places in the file

File: google/cloud/pubsub/internal/tracing_pull_ack_handler.h (hunk 4 above)
Comment (alevenberg) [reply google/cloud/pubsub/internal/tracing_pull_ack_handler.h]: removed

File: google/cloud/pubsub/internal/tracing_pull_ack_handler.h (hunk 3 above)
Comment (alevenberg) [reply google/cloud/pubsub/internal/tracing_pull_ack_handler.h]: removed

**Pull request merged**